pilot/pkg/networking/grpcgen/grpcgen.go

```diff
@@ -54,15 +54,37 @@ import (
 // to this value by the management server.
 const transportSocketName = "envoy.transport_sockets.tls"
 
-type GrpcConfigGenerator struct{}
+type GrpcConfigGenerator struct {
+	Cds model.XdsResourceGenerator
+}
+
+var shortnames = map[string]string{
+	v3.ListenerType: "lds",
+	v3.RouteType:    "rds",
+	v3.ClusterType:  "cds",
+}
+
+func clusterKey(hostname string, port int) string {
+	return subsetClusterKey("", hostname, port)
+}
+
+func subsetClusterKey(subset, hostname string, port int) string {
+	return model.BuildSubsetKey(model.TrafficDirectionOutbound, subset, host.Name(hostname), port)
+}
 
 func (g *GrpcConfigGenerator) Generate(proxy *model.Proxy, push *model.PushContext,
 	w *model.WatchedResource, updates *model.PushRequest) (model.Resources, model.XdsLogDetails, error) {
+	log.Infof("begininning gRPC  %s, gen for %s, names %v", shortnames[w.TypeUrl], proxy.ID, w.ResourceNames)
+
 	switch w.TypeUrl {
 	case v3.ListenerType:
 		return g.BuildListeners(proxy, push, w.ResourceNames), model.DefaultXdsLogDetails, nil
 	case v3.ClusterType:
-		return g.BuildClusters(proxy, push, w.ResourceNames), model.DefaultXdsLogDetails, nil
+		if g.Cds != nil {
```

Comment:
do we intend to have both paths long term?

Reply:
no, this is for experimentation

Reply:
Although it could be useful if someone only needs basic service discovery and has found a bug in the re-use path. Probably shouldn't be using in a non-experimental environment anyway.